CMake/RealmCore.cmake

```diff
@@ -390,8 +392,16 @@ macro(build_realm_sync)
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION_RELEASE ${sync_server_library_release})
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION ${sync_server_library_release})
 
-    find_package(PkgConfig)
-    pkg_check_modules(YAML QUIET yaml-cpp)
+    if(REALM_PLATFORM STREQUAL "Android")
+      ExternalProject_Add(YAML
+        GIT_REPOSITORY "git@github.com:jbeder/yaml-cpp.git"
+        CMAKE_ARGS "-DCMAKE_SYSTEM_NAME=Android -DCMAKE_ANDROID_ARCH_ABI=${ANDROID_ABI} -DCMAKE_ANDROID_NDK=${ANDROID_NDK}"
```

Comment:
I don't think so since AS is even shipping a modified version of cmake. But i think it is fine for object store to use 3.7, since will never build this alone in the AS.